addons/website/static/src/snippets/s_table_of_content/options.js

```diff
@@ -111,6 +111,11 @@ options.registry.TableOfContent = options.Class.extend({
         const exception = (tocEl) => !tocEl.querySelector('.s_table_of_content_navbar a');
         this._activateScrollSpy(exception);
         this.oldHeadingsEls = [...headingsEls.map(el => el.cloneNode(true))];
+        if (targetedEls.length === 0) {
+            // destroy public widget and remove the ToC since there are no more
+            // child elements.
+            this.trigger_up('remove_snippet', {$snippet: this.$target});
+        }
```

Comment:
I wonder if we cannot improve it a little bit more.. For now, if you delete the h1s and h2s of the block we remove it, even if there is still others texts in the block. I think it would be even better if the condition to remove the ToC block would be that there is no more text in it.
Something like that (to test)
```js
if (this.$target[0].textContent === "") {
    this.trigger_up("remove_snippet", {$snippet: this.$target});
}
```
What do you think ?

Reply:
Yes, absolutely 👍🏽